### PR TITLE
[#576] net-vpc - Add the ability to export/import custom routes throu…

### PR DESCRIPTION
**Pull request by LucaPrete**
…gh peering with PSA VPC

### CODE REVIEW

File: modules/net-vpc/README.md [hunk 1]
```diff
@@ -143,6 +143,28 @@ module "vpc" {
...
+  name       = "my-network"
+  subnets = [
+    {
+      ip_cidr_range      = "10.0.0.0/24"
+      name               = "production"
+      region             = "europe-west1"
+      secondary_ip_range = null
+    }
+  ]
+  psa_ranges               = {range-a = "10.10.0.0/16"}
```
Comment (ludoo) [inline modules/net-vpc/README.md:161]: Luca, we need to use a single variable for everything PSA, and allow individual control.

Something like

```hcl
psa_ranges = {
  foo = { range = "10.10.0.0/24", routes = null}
  bar = { range = "10.10.0.0/24", routes = {export = true, import = true}}
}
```

**Review by ludoo — CHANGES_REQUESTED** [review]

File: modules/net-vpc/README.md (hunk 1 above)
Comment (ludoo) [reply modules/net-vpc/README.md:161]: Please also use ranges that match with regular use cases in the examples (/24 or /28), which I did not in my comment above. :)

Comment (LucaPrete) [reply modules/net-vpc/README.md:161]: @ludoo thanks.
Actually, it's even worst... :)
One PSA range can have multiple IP ranges. Also, I would try to make the route import/export params more flat.

So, in my mind the structure would be:

```
psa_config = {
  foo = { ranges = ["10.10.0.0/24"]}
  bar = { ranges = ["192.168.1.0/24", "192.168.2.0/24"], export=true, import=true}
}
```

If we agree on the variables I can proceed with the implementation.

Comment (ludoo) [reply modules/net-vpc/README.md:161]: Well yes and no, keys are arbitrary (we need them to have static values for iteration) so we might as well keep one range per key. Unless n ranges map to one peering, then I agree with you as at that point the ke represents the peering.

Comment (LucaPrete) [reply modules/net-vpc/README.md:161]: Also in my case the answer is yes and no :)
It depends. Peerings are created per service per region.
So potentially yes, each key may represent a different peering, and yes: each Private Service Connection can use multiple ranges.

Comment (ludoo) [reply modules/net-vpc/README.md:161]: Hah :)

I would stick with one range per key so as to be less verbose. Unless you strongly feel for multiple ranges per key support.

Comment (LucaPrete) [reply modules/net-vpc/README.md:161]: ok, cool! That will make my life also easier when modifying the module. Thanks!

Comment (LucaPrete) [reply modules/net-vpc/README.md:161]: [for the records] - talking to Ludo, we've decided to allow multiple CIDR per PSA connection.

Final structure will be:

```
psa_config = {
  foo = { ranges = ["10.10.0.0/24"]}
  bar = { ranges = ["192.168.1.0/24", "192.168.2.0/24"], routes = {export=true, import=true}}
}
```

**Comment by LucaPrete** [comment]
This is now actually failing for checksum error downloading a terraform provider. Needs further checks. Then tests will (possibly) pass.

**Comment by sruffilli** [comment]
Re-ran checks - all green.

**Review by ludoo — APPROVED** [review]

**Pull request merged**